hazelcast/include/hazelcast/client/proxy/RingbufferImpl.h

```diff
@@ -153,9 +179,24 @@ namespace hazelcast {
                      * second ago, then there are 29 seconds remaining for that item. Using this policy you are going to overwrite no matter
                      * what.
                      */
-                     OVERWRITE = 0
+                            OVERWRITE = 0
                 };
 
+                static void checkSequence(int64_t sequence) {
```

Comment:
I think this method should be `checkNonNegativeSequence` or you could add this as a util method `checkNonNegative(T value, const string& message)` and use it in a lot of places since this will return IllegalArgumentException in most of the cases.

Reply:
We already have the Preconditions::checkNotNegative method. But this method here optimizes so that the error message string is not composed if the value is acceptable and this is a hot path.

Reply:
I see I am not sure how much of overhead that saves but it makes sense that it is faster. I would still say the method name should be `checkNonNegativeSequence ` because checkSequence does not say what it checks.